File: java/src/main/java/com/experian/dvs/client/phone/validate/PhoneType.java
package com.experian.dvs.client.phone.validate;

import java.util.HashMap;
import java.util.Map;

public enum PhoneType {
    MOBILE("Mobile"),
    LANDLINE("Landline"),
    MOBILE_OR_LANDLINE("Mobile or landline"),
    TOLL_FREE("Toll free"),
    PREMIUM("Premium"),
    SHARED_COST("Shared cost"),
    VOIP("VoIP"),
    STAGE_AND_SCREEN("Stage and screen"),
    PAGER("Pager"),
    UNIVERSAL_ACCESS_NUMBER("Universal access number"),
    PERSONAL_NUMBER("Personal number"),
    VOICEMAIL_ONLY("Voicemail only"),
    BAD_FORMAT("Bad format"),
    UNKNOWN("Unknown")
    ;

    private final String value;

    private static final Map<String, PhoneType> valueMap = new HashMap<>();

    static {
        for (PhoneType phoneValidationPhoneType : PhoneType.values()) {
            valueMap.put(phoneValidationPhoneType.value, phoneValidationPhoneType);
        }
    }

    public static PhoneType fromValue(String value) {
        return valueMap.get(value);
    }

    PhoneType(String value) {
        this.value = value;
    }
}

File: typescript/src/phone/validate/phoneType.ts
export enum PhoneType {
    Mobile = "Mobile",
    Landline = "Landline",
    MobileOrLandline = "Mobile or landline",
    TollFree = "Toll free",
    Premium = "Premium",
    SharedCost = "Shared cost",
    VoIP = "VoIP",
    StageAndScreen = "Stage and screen",
    Pager = "Pager",
    UniversalAccessNumber = "Universal access number",
    PersonalNumber = "Personal number",
    VoicemailOnly = "Voicemail only",
    BadFormat = "Bad format",
    Unknown = "Unknown"
}

export function lookupPhoneType(str?: string): PhoneType {
    if (str) {
        return Object.keys(PhoneType).find(key => PhoneType[key as keyof typeof PhoneType] === str) as PhoneType | PhoneType.Unknown;    
    } else {
        return PhoneType.Unknown;
    }
}
